validation-test/stdlib/FixedPointDiagnostics.swift.gyb

```diff
@@ -2,13 +2,13 @@
 // RUN: %line-directive %t/main.swift -- %target-swift-frontend -parse -verify %t/main.swift
 
 func testUnaryMinusInUnsigned() {
-  var a: UInt8 = -(1) // expected-error {{}} expected-note * {{}}
+  var a: UInt8 = -(1) // expected-error {{}} expected-note * {{}} expected-warning * {{}}
```

Comment:
Just figured out what is going on with this warning, and here it is - since this is being diagnosed as contextual conversion error, actual pattern_binding declaration for "a" is going to be type-checked as "Int" because `typeCheckChildIndependently` applies solutions to AST...

Before changes (because "+" operation itself has been diagnosed to the type is not set):
```
(brace_stmt
  (pattern_binding_decl
    (pattern_typed type='UInt8'
      (pattern_named type='UInt8' 'a')
      (type_ident
        (component id='UInt8' bind=Swift.(file).UInt8)))
    (prefix_unary_expr type='<<error type>>' location=<REPL Input>:2:19 range=[<REPL Input>:2:19 - line:2:22]
      (overloaded_decl_ref_expr type='<<error type>>' location=<REPL Input>:2:19 range=[<REPL Input>:2:19 - line:2:19] name=- #decls=7 specialized=no function_ref=unapplied
      Swift.(file).-
      Swift.(file).-
      Swift.(file).-
      Swift.(file).-
      Swift.(file).-
      Swift.(file).-
      Swift.(file).SignedNumber.-)
      (paren_expr type='<<error type>>' location=<REPL Input>:2:21 range=[<REPL Input>:2:20 - line:2:22]
        (integer_literal_expr type='<<error type>>' location=<REPL Input>:2:21 range=[<REPL Input>:2:21 - line:2:21] value=1))))

  (var_decl "a" type='UInt8' access=private storage_kind=stored)
)
```
 After changes (because "+" is considered valid and contextual conversion is invalid):

```
(brace_stmt
  (pattern_binding_decl
    (pattern_typed type='UInt8'
      (pattern_named type='UInt8' 'a')
      (type_ident
        (component id='UInt8' bind=Swift.(file).UInt8)))
    (prefix_unary_expr type='Int' location=<REPL Input>:2:19 range=[<REPL Input>:2:19 - line:2:22]
      (declref_expr type='(Int) -> Int' location=<REPL Input>:2:19 range=[<REPL Input>:2:19 - line:2:19] decl=Swift.(file).- [with Int[Int: SignedNumber module Swift], Int] function_ref=unapplied specialized=no)
      (paren_expr type='(Int)' location=<REPL Input>:2:21 range=[<REPL Input>:2:20 - line:2:22]
        (call_expr implicit type='Int' location=<REPL Input>:2:21 range=[<REPL Input>:2:21 - line:2:21]  arg_labels=_builtinIntegerLiteral:
          (constructor_ref_call_expr implicit type='(Int2048) -> Int' location=<REPL Input>:2:21 range=[<REPL Input>:2:21 - line:2:21]
            (declref_expr implicit type='(Int.Type) -> (Int2048) -> Int' location=<REPL Input>:2:21 range=[<REPL Input>:2:21 - line:2:21] decl=Swift.(file).Int.init(_builtinIntegerLiteral:) function_ref=single specialized=no)
            (type_expr implicit type='Int.Type' location=<REPL Input>:2:21 range=[<REPL Input>:2:21 - line:2:21] typerepr='Int'))
          (tuple_expr implicit type='(_builtinIntegerLiteral: Int2048)' location=<REPL Input>:2:21 range=[<REPL Input>:2:21 - line:2:21] names=_builtinIntegerLiteral
            (integer_literal_expr type='Int2048' location=<REPL Input>:2:21 range=[<REPL Input>:2:21 - line:2:21] value=1))))))

  (var_decl "a" type='UInt8' access=private storage_kind=stored)
)
```

This looks to be caused by the fact that `diagnoseContextualConversionError` doesn't erase type of the expression after it being diagnosed as erroneous...

Do you think I should try and address it in that patch?